luigi_pipeline/tests/test_seqr_gcnv_loading.py

```diff
@@ -0,0 +1,386 @@
+                hl.Struct(genes="", gene_set=set()),
+                hl.Struct(genes="SLC35A4.43", gene_set=set(["SLC35A4"])),
+            ],
+        )
+
+    def test_aggregate_parsed_genes(self):
+        t1 = hl.Table.parallelize(
+            [
+                {"genes": "AC118553.2,SLC35A3"}, 
+                {"genes": "AC118553.1,None"}, 
+                {"genes": "None"},
+                {"genes": "SLC35A3.43"}, 
+                {"genes": ""}, 
+                {"genes": "SLC35A4.43"},
+            ], 
+            hl.tstruct(genes=hl.dtype('str')), 
+            key="genes"
+        )
+        aggregated_gene_set = t1.aggregate(
+            hl_agg_collect_set_union(parse_genes(t1.genes))
+        )
+        self.assertEqual(
+            aggregated_gene_set,
+            set(["SLC35A4", "SLC35A3", "AC118553"])
+        )
+
+class SeqrGCNVLoadingTest(unittest.TestCase):
+    maxDiff = None
+
+    def setUp(self):
+        self._temp_dir = tempfile.TemporaryDirectory()
+        self._variant_mt_file = tempfile.mkstemp(dir=self._temp_dir.name, suffix='.mt')[1]
+        self._genotypes_mt_file = tempfile.mkstemp(dir=self._temp_dir.name, suffix='.mt')[1]
+
+    def tearDown(self):
+        shutil.rmtree(self._temp_dir.name)
+
+    @mock.patch('lib.model.gcnv_mt_schema.datetime', wraps=datetime)
+    def test_run_new_joint_tsv_task(self, mock_datetime):
+        mock_datetime.date.today.return_value = datetime.date(2022, 12, 2)
+        worker = luigi.worker.Worker()
+        variant_task = SeqrGCNVVariantMTTask(
+            source_paths=NEW_JOINT_CALLED_CALLSET,
+            dest_path=self._variant_mt_file,
+        )
+        genotype_task = SeqrGCNVGenotypesMTTask(
+            genome_version="38",
+            source_paths="i am completely ignored",
+            dest_path=self._genotypes_mt_file,
+            is_new_joint_call=True,
+        )
+        SeqrGCNVGenotypesMTTask.requires = lambda self: [variant_task]
+        worker.add(genotype_task)
+        worker.run()
+
+        variant_mt = hl.read_matrix_table(self._variant_mt_file)
+        self.assertEqual(variant_mt.count(), (3, 4))
+
+        key_dropped_variant_mt = variant_mt.rows().flatten().drop("variant_name", "svtype")
+        data = key_dropped_variant_mt.collect()
+        self.assertCountEqual(data, NEW_JOINT_CALLED_EXPECTED_VARIANT_DATA)
+
+        # Genotypes (only) Assertions
+        genotypes_mt = hl.read_matrix_table(self._genotypes_mt_file)
+        self.assertEqual(genotypes_mt.count(), (3, 4))
+
+        # Now mimic the join in BaseMTToESOptimizedTask
+        genotypes_mt = genotypes_mt.drop(*[k for k in genotypes_mt.globals.keys()])
+        row_ht = genotypes_mt.rows().join(variant_mt.rows()).flatten().drop("variant_name", "svtype")
+        data = row_ht.collect()
+        for i, row in enumerate(data):
+            data[i] = prune_empties(row)
+        self.assertListEqual(data, NEW_JOINT_CALLED_EXPECTED_VARIANT_AND_GENOTYPES_DATA)
```

Comment:
I'm curious if it is better to mock the `HailElasticsearchClient` and test the parameters for calling `export_table_to_elasticsearch`.